torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1439,17 +1429,17 @@ void ProcessGroupNCCL::ncclCommWatchdog() {
   } catch (std::exception& e) {
     if (std::string(e.what()).find("driver shutting down") !=
         std::string::npos) {
-      LOG(INFO)
-          << "[Rank " << rank_
-          << "] main process destroyed cuda before watchdog loop exited, terminating watchdog."
-          << " (Watchdog caught exception: " << e.what();
-
+      LOG(INFO) << fmt::format(
+          "[Rank {}] "
+          "main process destroyed cuda before watchdog loop exited, terminating watchdog. "
+          "(Watchdog caught exception: {})",
+          rank_,
+          e.what());
```

Comment:
Just curious: why one is better than another? (I.e. I understand the desire of getting rid of `c10::str` but this would be downright slower, wouldn't it?)